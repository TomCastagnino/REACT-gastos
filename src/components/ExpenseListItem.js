import React from 'react';
import {Link} from 'react-router-dom';
import moment from 'moment';
import numeral from 'numeral';

const ExpenseListItem = props => (
    <div>
        <p>
            <Link 
                to={`/edit/${props.expense.id}`} 
            >
                {props.expense.description}
            </Link>, amount: {numeral(props.expense.amount / 100).format('$0,0.00')}, created: {moment(props.expense.createdAt).format('Do MMMM, YYYY')}.
        </p>
    </div>
);

export default ExpenseListItem;