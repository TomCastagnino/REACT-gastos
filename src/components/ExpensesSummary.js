import React from 'react';
import {connect} from 'react-redux';
import selectExpensesTotal from '../selectors/expenses-total';
import getVisibleExpenses from '../selectors/expenses';
import numeral from 'numeral';

export const ExpensesSummary = ({expensesCount, expensesTotal}) => {
    return (
        <div>
            <p>Viewing {expensesCount} expense{`${expensesCount === 1 ? '' : 's'}`} totalling: {numeral(expensesTotal / 100).format('$0,0.00') }</p>
        </div>
    );
}

const mapStateToProps = state => {
    const visibleExpenses = getVisibleExpenses(state.expenses, state.filters)
    return {
        expensesCount: visibleExpenses.length,
        expensesTotal: selectExpensesTotal(visibleExpenses)
    };
};

export default connect(mapStateToProps)(ExpensesSummary);