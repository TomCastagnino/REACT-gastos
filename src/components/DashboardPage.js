import React from 'react';
import ExpenseList from './ExpenseList';
import ExpensesSummary from './ExpensesSummary';
import ExpenseListFilters from './ExpenseListFilters';

const DashboardPage = () => (
    <div>
        <ExpenseListFilters />
        <ExpensesSummary />
        <ExpenseList /> 
    </div>
);

export default DashboardPage;