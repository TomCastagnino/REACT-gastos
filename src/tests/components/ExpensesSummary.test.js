import React from 'react';
import {shallow} from 'enzyme';
import {ExpensesSummary} from '../../components/ExpensesSummary';

test('Should render ExpenseSummary', () => {
    const wrapper = shallow(<ExpensesSummary expensesCount={1} expensesTotal={100} />);
    expect(wrapper).toMatchSnapshot();
});

test('Should render empty ExpenseSummary', () => {
    const wrapper = shallow(<ExpensesSummary expensesCount={15} expensesTotal={100643} />);
    expect(wrapper).toMatchSnapshot();
});