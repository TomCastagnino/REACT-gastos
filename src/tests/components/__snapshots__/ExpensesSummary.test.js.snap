// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render ExpenseSummary 1`] = `
<div>
  <p>
    Viewing 
    1
     expense
     totalling: 
    $1.00
  </p>
</div>
`;

exports[`Should render empty ExpenseSummary 1`] = `
<div>
  <p>
    Viewing 
    15
     expense
    s
     totalling: 
    $1,006.43
  </p>
</div>
`;
